pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -623,10 +698,15 @@ cdef class {{name}}HashTable(HashTable):
 
         with nogil:
             for i in range(n):
-                val = values[i]
+                val = {{to_c_type}}(values[i])
 
                 # specific for groupby
-                {{if dtype != 'uint64'}}
+                {{if dtype == 'complex64' or dtype== 'complex128'}}
+                # TODO: what should be done here?
```

Comment:
Hmm... this is probably a question for floats and a bug for signed types with less than 64bits